catalog/security/catalog-security-policyplugin/src/main/java/org/codice/ddf/catalog/security/CatalogPolicy.java

```diff
@@ -0,0 +1,274 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.security;
+
+import java.io.Serializable;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.Result;
+import ddf.catalog.operation.Query;
+import ddf.catalog.operation.ResourceRequest;
+import ddf.catalog.operation.ResourceResponse;
+import ddf.catalog.plugin.PolicyPlugin;
+import ddf.catalog.plugin.PolicyResponse;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.plugin.impl.PolicyResponseImpl;
+import ddf.catalog.util.impl.Requests;
+
+/**
+ * IngestPlugin is a PreIngestPlugin that restricts the create/update/delete operations
+ * on the catalog to a group defined by a set of configurable user attributes.
+ */
+public class CatalogPolicy implements PolicyPlugin {
+
+    private String[] createPermissions;
+
+    private String[] updatePermissions;
+
+    private String[] deletePermissions;
+
+    private String[] readPermissions;
+
+    private Map<String, Set<String>> createPermissionMap = new HashMap<>();
```

Comment:
Instead of `Map<String, Set<String>>`, should we create a permission interface and hide this implementation detail?